junit-engine-api/src/main/java/org/junit/gen5/engine/EngineExecutionListener.java

```diff
@@ -10,12 +10,17 @@
 
 package org.junit.gen5.engine;
 
+import java.util.*;
+
 /**
  * @since 5.0
  */
 public interface EngineExecutionListener {
 
-	void dynamicTestRegistered(TestDescriptor testDescriptor);
+	void reportingEntryPublished(TestDescriptor testDescriptor, Map<String, String> entry);
```

Comment:
I think `entry` should actually be some kind of class to wrap the key-value pairs and maybe add something like a timestamp? This would also simplify evolution.